web-common/src/features/canvas/stores/canvas-spec.ts

```diff
@@ -1,24 +1,199 @@
-import type { QueryObserverResult } from "@rilldata/svelte-query";
-import { type CanvasResponse } from "@rilldata/web-common/features/canvas/selector";
-import type {
-  RpcStatus,
-  V1CanvasSpec,
+import type { CanvasSpecResponseStore } from "@rilldata/web-common/features/canvas/types";
+import {
+  MetricsViewSpecMeasureType,
+  type MetricsViewSpecDimensionV2,
+  type MetricsViewSpecMeasureV2,
+  type V1CanvasSpec,
 } from "@rilldata/web-common/runtime-client";
-import { derived, type Readable } from "svelte/store";
+import { derived, get, type Readable } from "svelte/store";
 
 export class CanvasResolvedSpec {
   canvasSpec: Readable<V1CanvasSpec | undefined>;
   isLoading: Readable<boolean>;
```

Comment:
Do we need `isError` and `error` for error handling?

Reply:
If so, might it be easier to expose the `$validSpecStore` itself, rather than duplicating the fields that map 1:1 to the TanStack Query response? If so, maybe the exposure of the `useCanvas` response belongs in the top-level `CanvasEntity` class?